### PR TITLE
Fix configuration-entries for bg-deploy

### PR DESCRIPTION
**Pull request by boyan-velinov**
Jira:LMCROSSITXSADEPLOY-915

### CODE REVIEW

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/RebuildApplicationDeployModelStep.java [hunk 1]
```diff
@@ -71,4 +104,14 @@ private CloudApplicationExtended findApplication(List<CloudApplicationExtended>
             .get();
     }
```
Comment (KristianShishoev) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/RebuildApplicationDeployModelStep.java]: remove comments.

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PrepareAppsDeploymentStep.java [hunk 2]
```diff
@@ -50,6 +55,13 @@ protected StepPhase executeStep(ExecutionWrapper execution) {
         execution.getContext()
             .setVariable(Constants.EXECUTE_ONE_OFF_TASKS, true);
```
Comment (KristianShishoev) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PrepareAppsDeploymentStep.java]: Better way in my opinion is to delete if statement and assign direct  this statement 
 ProcessType.BLUE_GREEN_DEPLOY.equals(processType) to skipUpdateConfigurationEntries variable.
Make your choice :)

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/message/Messages.java [hunk 3]
```diff
@@ -80,6 +80,7 @@
     public static final String ERROR_RESOLVING_DESCRIPTOR_PROPERTIES = "Error resolving merged descriptor properties and parameters";
     public static final String ERROR_CREATING_SUBSCRIPTIONS = "Error creating subscriptions";
     public static final String ERROR_BUILDING_CLOUD_MODEL = "Error building cloud model";
+    public static final String ERROR_RESOLVING_CONFIGURATION_ENTRIES = "Error building cloud model";
```
Comment (enchobelezirev) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/message/Messages.java]: Duplicate message?

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/message/Messages.java [hunk 4]
```diff
@@ -264,6 +266,7 @@
     public static final String SCALING_APP = "Scaling application \"{0}\"...";;
     public static final String APP_SCALED = "Application \"{0}\" scaled";
     public static final String PUBLISHING_PUBLIC_PROVIDED_DEPENDENCIES = "Publishing public provided dependencies for application \"{0}\"...";
+    public static final String NO_PUBLIC_PROVIDED_DEPENDENCIES = "No public provided dependencies";
```
Comment (enchobelezirev) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/message/Messages.java]: Why is that?

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/BuildCloudDeployModelStep.java [hunk 5]
```diff
@@ -100,6 +103,23 @@ protected StepPhase executeStep(ExecutionWrapper execution) throws SLException {
         }
     }
 
+    private void buildConfigurationEntriesToPublish(DelegateExecution context, DeploymentDescriptor deploymentDescriptor,
+        List<CloudApplicationExtended> apps) {
+        ProcessType processType = processTypeParser.getProcessType(context);
+
+        if (processType.equals(ProcessType.BLUE_GREEN_DEPLOY)) {
+            StepsUtil.setConfigurationEntriesToPublish(context, Collections.emptyMap());
+            StepsUtil.setSkipUpdateConfigurationEntries(context, true);
```
Comment (enchobelezirev) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/BuildCloudDeployModelStep.java]: Could we remove this skipUpdateConfigurationEntries variable and calculate whether it should be true or false based on the configurationEntriesToPublish?

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PublishConfigurationEntriesStep.java [hunk 6]
```diff
@@ -8,6 +8,7 @@
 
 import javax.inject.Inject;
 
+import org.apache.commons.collections.MapUtils;
```
Comment (enchobelezirev) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PublishConfigurationEntriesStep.java]: Why using this dep?

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/ResolveConfigurationEntriesStep.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMta;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaModule;
+import com.sap.cloud.lm.sl.cf.core.util.CloudModelBuilderUtil;
+import com.sap.cloud.lm.sl.cf.process.message.Messages;
+import com.sap.cloud.lm.sl.common.SLException;
+import com.sap.cloud.lm.sl.mta.model.v1_0.DeploymentDescriptor;
+
+@Component("resolveConfigurationEntriesStep")
+@Scope(BeanDefinition.SCOPE_PROTOTYPE)
+public class ResolveConfigurationEntriesStep extends BuildCloudDeployModelStep {
```
Comment (enchobelezirev) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/ResolveConfigurationEntriesStep.java]: I suppose the logic here is taken from the CloudModelBuilderStep?
Where this step is being used? If it is called each time, why not remove the logic from the CloudModelBuilder?

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/RebuildApplicationDeployModelStep.java (hunk 1 above)
Comment (boyan-velinov) [reply com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/RebuildApplicationDeployModelStep.java]: done

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PrepareAppsDeploymentStep.java (hunk 2 above)
Comment (boyan-velinov) [reply com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PrepareAppsDeploymentStep.java]: Agree.

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/message/Messages.java (hunk 3 above)
Comment (boyan-velinov) [reply com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/message/Messages.java]: Not needed => removed

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/message/Messages.java (hunk 4 above)
Comment (boyan-velinov) [reply com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/message/Messages.java]: For debug message

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/BuildCloudDeployModelStep.java (hunk 5 above)
Comment (boyan-velinov) [reply com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/BuildCloudDeployModelStep.java]: Actually, configurationEntriesToPublish are build depending if configuration entries will be updated or skiped.
In parent process in PrepareAppsDeploymentStep StepsUtil.setSkipUpdateConfigurationEntries is invoked.
In sub-process in RebuildApplicationDeployModelStep configurationEntries are calculated based on "skip" flag.

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PublishConfigurationEntriesStep.java (hunk 6 above)
Comment (boyan-velinov) [reply com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PublishConfigurationEntriesStep.java]: Not needed => removed

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/ResolveConfigurationEntriesStep.java (hunk 7 above)
Comment (boyan-velinov) [reply com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/ResolveConfigurationEntriesStep.java]: It is changed in the next commit

**Pull request merged**